packages/composites/the-graph/src/methods/prices/prices.ts

```diff
@@ -16,111 +19,161 @@ const customParams = {
-    referenceContract,
-    referenceContractDivisor,
-    referenceModifierAction: referenceModifierAction.toUpperCase() as ReferenceModifierAction,
-    intermediaryToken: intermediaryToken.toUpperCase(),
-  }
-  if (baseCoinTicker === quoteCoinTicker) {
-    throw new Error('Base and Quote coins must be different')
-  }
-  Logger.info(`Fetching quote for ${quoteCoinTicker}/${baseCoinTicker} pair from ${dex}`)
-  let price
-  try {
-    price = await getQuotePrice(inputParams, dexSubgraph)
-  } catch (e) {
-    throw new Error(`Failed to get price.  Reason "${e}"`)
-  }
-  return Requester.success(
-    jobRunID,
-    {
-      status: 200,
-      data: {
-        result: price,
-      },
-    },
-    true,
-  )
+export const execute = async (input: AdapterRequest, config: Config): Promise<AdapterResponse> => {
+    const validator = new Validator(input, customParams)
+    if (validator.error) throw validator.error
+    const jobRunID = validator.validated.id
+    const inputParams = validator.validated.data
+    if (!inputParams.theGraphQuote && !inputParams.quoteCoinTicker) {
+        throw new Error("quoteCoinTicker cannot be empty if theGraphQuote not supplied")
+    }
+    const dexToUpperCase = inputParams.dex.toUpperCase()
+    const dexSubgraph = config.dexSubgraphs[dexToUpperCase]
+    if (!dexSubgraph) {
+        throw new Error(`${inputParams.dex} is currently not supported`)
+    }
+    let price
+    switch(dexToUpperCase) {
+        case UNISWAP:
+            price = await executeUniswap(jobRunID, inputParams, dexSubgraph as UniswapSubgraph)
+            break;
+        case CURVE:
+            price = await executeCurve(jobRunID, inputParams, dexSubgraph as CurveSubgraph)
+            break;
+        default:
+            price = await executeUniswap(jobRunID, inputParams, dexSubgraph as UniswapSubgraph)
+            break;
+    }
+    return Requester.success(jobRunID, {
+        status: 200,
+        data: {
+            result: price
+        }
+    }, true)
 }
 
-const getQuotePrice = async (
-  inputParams: DexQueryInputParams,
-  dexSubgraph: DexSubgraph,
-): Promise<number> => {
-  const { jobRunID, baseCoinTicker, quoteCoinTicker, referenceContract } = inputParams
-  const token0 = await dexSubgraph.getToken(jobRunID, baseCoinTicker)
-  const token1 = await dexSubgraph.getToken(jobRunID, quoteCoinTicker)
-  let token1PerToken0 = await dexSubgraph.getTokenPairPrice(jobRunID, token0.id, token1.id)
-  if (!token1PerToken0) {
-    token1PerToken0 = await getPriceThroughCommonPair(
-      inputParams,
-      dexSubgraph,
-      token0.id,
-      token1.id,
-    )
-  }
-  Logger.info(`Price of ${quoteCoinTicker}/${baseCoinTicker} is ${token1PerToken0}`)
-  if (referenceContract) {
-    token1PerToken0 = await modifyResultByFeedResult(inputParams, token1PerToken0)
-  }
-  return token1PerToken0
+export const executeCurve = async(jobRunID: string, input: DexQueryInputParams, dexSubgraph: CurveSubgraph): Promise<Decimal> => {
```

Comment:
Can we move this logic into `curve.ts` and import it here?  I'd like to keep dex specific logic out of this file.